spectator-reg-atlas/src/test/java/com/netflix/spectator/atlas/AtlasTimerTest.java

```diff
@@ -144,6 +145,116 @@ public void recordSeveralValues() {
     checkValue(4, 1 + 2 + 3 + 1, 1 + 4 + 9 + 1, 3);
   }
 
+  @Test
+  public void recordBatchMismatchedLengths() {
+    dist.record(new long[0], 1, TimeUnit.NANOSECONDS);
+    clock.setWallTime(1 * step + 1);
+    checkValue(0, 0, 0, 0);
+
+    dist.record(new long[1], 0, TimeUnit.NANOSECONDS);
+    clock.setWallTime(2 * step + 1);
+    checkValue(0, 0, 0, 0);
+
+    dist.record(new long[1], -1, TimeUnit.NANOSECONDS);
+    clock.setWallTime(3 * step + 1);
+    checkValue(0, 0, 0, 0);
+
+    dist.record(new long[]{ 0, 0 }, 2, TimeUnit.NANOSECONDS);
+    clock.setWallTime(4 * step + 1);
+    checkValue(2, 0, 0, 0);
+  }
+
+  @Test
+  public void recordBatchZero() {
+    dist.record(new long[]{ 0 }, 1, TimeUnit.NANOSECONDS);
+    checkValue(0, 0, 0, 0);
+
+    clock.setWallTime(step + 1);
+    checkValue(1, 0, 0, 0);
+  }
+
+  @Test
+  public void recordBatchOne() {
+    dist.record(new long[]{ 1 }, 1, TimeUnit.NANOSECONDS);
+    checkValue(0, 0, 0, 0);
+
+    clock.setWallTime(step + 1);
+    checkValue(1, 1, 1, 1);
+  }
+
+  @Test
+  public void recordBatchTwo() {
+    dist.record(new long[]{ 2 }, 1, TimeUnit.NANOSECONDS);
+    checkValue(0, 0, 0, 0);
+
+    clock.setWallTime(step + 1);
+    checkValue(1, 2, 4, 2);
+  }
+
+
+  @Test
+  public void recordBatchOverflowSingle()
+  {
```

Comment:
formatting, please move brace to end of previous line to be consistent with other code